flax/linen/attention.py

```diff
@@ -112,9 +112,9 @@ def dot_product_attention_weights(query: Array,
     if broadcast_dropout:
       # dropout is broadcast across the batch + head dimensions
       dropout_shape = tuple([1] * (key.ndim - 2)) + attn_weights.shape[-2:]
-      keep = random.bernoulli(dropout_rng, keep_prob, dropout_shape)
+      keep = random.bernoulli(dropout_rng, keep_prob, dropout_shape) # type: ignore
```

Comment:
same Q here: how was jax breaking typechecks here?

Reply:
`Argument 1 to "bernoulli" has incompatible type "Optional[Any]"; expected "Union[Array, PRNGKeyArray]"`

Basically `PRNGKey` is defined differently in flax and jax, and bringing the two in line would have involved much larger changes.